File: solutions/elixir/resistor-color/2/lib/resistor_color.ex
defmodule ResistorColor do
  @doc """
  Return the value of a color band
  """

  @map_color_to_value %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }
  
  @spec code(atom) :: integer()
  def code(color) do
    Map.get(@map_color_to_value, color, nil)
  end
end

File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

// Colors returns the list of all colors.
func Colors() []string {
	resistors := []string {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    }
    return resistors
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	if color == "black" {
        return 0
    } else if color == "brown" {
        return 1
    } else if color == "red" {
        return 2
    } else if color == "orange" {
        return 3
    } else if color == "yellow" {
        return 4
    } else if color == "green" {
        return 5
    } else if color == "blue" {
        return 6
    } else if color == "violet" {
        return 7
    } else if color == "grey" {
        return 8
    } else /*if color == "white"*/ {
        return 9
    }
}
